.github/workflows/publish-tag.yml

```diff
@@ -0,0 +1,102 @@
+name: Rust build, benchmarks, and tests
+
+concurrency:
+  group: ci-${{ github.ref }}
+  cancel-in-progress: true
+
+##
+on:
+  ##
+  # Run when a semantic version is tagged
+  push:
+    tags:
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+
+##
+# Environment variables shared for all targets
+env:
+  CARGO_TERM_COLOR: always
+  RELEASE: true
+  RELEASE_NAME: release
+  RUSTFLAGS: -A warnings
+  RUST_BACKTRACE: full
+  SKIP_WASM_BUILD: 1
+  VERBOSE: ${{ github.events.input.verbose }}
+
+##
+# Test and build and publish
+jobs:
+  check:
+    name: Tests and publishes targeting ${{ matrix.rust-target }} for OS ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+
+    ##
+    # Define multiple targets for builds and tests
+    strategy:
+      matrix:
+        rust-branch:
+          - stable
+
+        rust-target:
+          - x86_64-unknown-linux-gnu
+          - x86_64-apple-darwin
+
+        os:
+          - ubuntu-latest
+          - macos-latest
+
+        include:
+          - os: ubuntu-latest
+          - os: macos-latest
+
+    ##
+    # Environment variables specific to each target
+    env:
+      RUSTV: ${{ matrix.rust-branch }}
+      RUST_BIN_DIR: target/${{ matrix.rust-target }}/release
+      TARGET: ${{ matrix.rust-target }}
+
+    ##
+    steps:
+      - name: Check-out repository under $GITHUB_WORKSPACE
+        uses: actions/checkout@v2
+
+      - name: Install dependencies
+        run: |
+          sudo apt update &&
+          sudo apt install -y git clang curl libssl-dev llvm libudev-dev protobuf-compiler
+
+      - name: Install Rust ${{ matrix.rust-branch }}
+        uses: actions-rs/toolchain@v1.0.6
+        with:
+          toolchain: ${{ matrix.rust-branch }}
+          profile: minimal
+
+      - name: Utilize Rust shared cached
+        uses: Swatinem/rust-cache@v2.2.1
+        with:
+          key: ${{ matrix.os }}-${{ env.RUST_BIN_DIR }}
+
+      - name: Run tests
+        run: |
+          cargo test --tests
+
+      - name: Build executable
+        run: |
+          cargo build --release
+
+      ## TODO: double-check `artifacts` path(s) be correct
+      # :warning: values for the following must always match;
+      #   - RUST_BIN_DIR
+      #   - target/${{ matrix.rust-target }}/release/node-subtensor
+      - name: Create Release
+        uses: ncipollo/release-action@v1.12.0
+        with:
+          allowUpdates: false
+          artifactErrorsFailBuild: true
+          bodyFile: CHANGELOG.md
```

Comment:
what is this for?

It is going to read from that file or to write the github release there aswell??

Reply:
Yup!... According to the [`ncipollo/release-action`][1] Action;

> "An optional body file for the release. This should be the path to the file."

... Effectively the file will need to be updated as part of a Git Tag check-list, because that'll be published as the content for the GitHub Release, eg.

- [ ] Bump the correct [semantic version][2] component
- [ ] Update the `CHANGELOG.md` file with any relevant notes and/or references
- [ ] Tag via CLI `git tag --annotate vM.I.P --message 'Spiffy code-name'`
- [ ] Push via CLI `git push origin vM.I.P`
- [X] :robot: CI/CD takes over here to accept or reject release

[1]: https://github.com/ncipollo/release-action
[2]: https://semver.org/